test/integration/integration_test.py

```diff
@@ -3,26 +3,30 @@
 import logging
 import os
 import sys
+import multiprocessing
 import pytest
 
 from test.integration import utility
 
 if __name__ == '__main__':
   path = os.path.dirname(os.path.realpath(__file__))
   test_selection_arg = sys.argv[1] if len(sys.argv) > 1 else ""
+  num_cores = multiprocessing.cpu_count()
+  num_workers = max(1, min(num_cores - 1, 4 if utility.isSanitizerRun() else num_cores))
```

Comment:
assuming that you dont want to actually test the kernels abitlity to context switch - this shouldnt be set to more than the actual cores available